client/src/components/AddItem.js

```diff
@@ -72,20 +77,43 @@ const AddItem = () => {
     return response.json();
   };
 
+  const getLists = async () => {
+    const res = await fetch(`/lists?user_id=${userId}`);
+    const json = await res.json();
+    setUserLists(json);
+  };
+
+  const getListId = () => {
+    userLists.forEach(function (list) {
+      if (list.name == selectedList) {
+        setListId(list.id);
+      }
+    });
+  };
+
   const addButtonClick = async () => {
     // TODO: regex check inputLink here is a valid URL to scrape
     if (inputLink.length > 0) {
       openPopup();
       const newItem = await getItem(inputLink);
       setItem(newItem);
-    }
+    } 
   };
 
   const closePopup = () => {
     setPopupOpen(false);
     setItem({});
   };
 
+  useEffect(() => {
+    getLists();
+  }, []);
+
+
+  useEffect(() => {
+    getListId();
+  }, [selectedList]);
+
```

Comment:
Rather than using a useEffect, we just want a function like changeList that get calls when you pick a different one on the drop down. Also, rather than passing the id  of the list, that function should be passed the index of the list being chosen. This will not only be more efficient but will also handle the use case where a user happens to name two lists the same thing